azkaban-common/src/main/java/azkaban/project/DirectoryFlowLoader.java

```diff
@@ -135,14 +138,25 @@ public void loadProjectFlow(final Project project, final File baseDirectory) {
     this.rootNodes = new HashSet<>();
     this.flowDependencies = new HashMap<>();
 
-    // Load all the props files and create the Node objects
-    loadProjectFromDir(baseDirectory.getPath(), baseDirectory, null);
+    if(azkabanFlowVersion == null) {
+      // Load all the props files and create the Node objects
+      loadProjectFromDir(baseDirectory.getPath(), baseDirectory, null);
+      // Create edges and find missing dependencies
+      resolveDependencies();
+    }
+    else if(azkabanFlowVersion.equals(Constants.AZKABAN_FLOW_VERSION_2_0)) {
+      // For azkaban.flow.2.0, yaml files will be used to describe flow info.
+      // First convert yaml files to azkabanFlows, and then load project from
+      // azkabanFlows. This is the first step to integrate azkabanFlows in the
+      // core azkaban code.
+      for (AzkabanFlow azkabanFlow : convertYamlFileToAzkabanFlows(baseDirectory)) {
+        loadProjectFromAzkabanFlow(azkabanFlow);
+      }
+      resolveDependenciesFromAzkabanFlow();
```

Comment:
better to warn here if it hits else block

Reply:
Agree, thanks!